pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -220,6 +218,10 @@ in stdenv.mkDerivation {
 
   doCheck = false; # fails a bunch of tests
 
+  preInstall = ''
+    export DESTDIR=/
```

Comment:
What error does this line fix?

Reply:
Afaik it should be already the default `DESTDIR`.

Reply:
No, it wasn't, and by setting it, we can avoid having to patch things inside the build system manually (see the removed patches inhttps://github.com/NixOS/nixpkgs/pull/85334/commits/0465dd7b85e3cf6f63f405df2ce66b1b901d4b36).

Reply:
Could you add a comment with explanation?